File: Object-oriented software design/Solutions/C/LC/E1/LocalFactory.cs
using System;

namespace E1 {
	public class LocalFactory {
		public static Func<ORM> ORMProvider { get; set; }

		public ORM CreateORM() {
			if (ORMProvider == null)
				throw new Exception("No ORM provider");

			return ORMProvider();
		}
	}
}

File: Object-oriented software design/Solutions/C/E3/src/com/company/LocalFactory.java
package com.company;

import java.util.concurrent.Callable;

public class LocalFactory {
    private static Callable<ORM> ORMProvider;

    public ORM CreateORM() throws Exception {
        if (ORMProvider == null)
            throw new Exception("No ORM provider");

        return ORMProvider.call();
    }

    public static Callable<ORM> getORMProvider() {
        return ORMProvider;
    }

    public static void setORMProvider(Callable<ORM> ORMProvider) {
        LocalFactory.ORMProvider = ORMProvider;
    }
}
